services/UserAccountService.ts

```diff
@@ -225,13 +225,23 @@ export default class UserAccountService {
         return map;
       }, {});
 
-      const updatedUsers = await Promise.all(accessUpdates.map(async (accessUpdate, index) => {
+      const updatedUsers = await Promise.all(accessUpdates.map(async (accessUpdate) => {
         const { user: userEmail, accessType } = accessUpdate;
 
-        const currUser = emailToUserMap[userEmail];
-        const oldAccess = currUser.accessType;
+        // Prevent a user from demoting themselves
+        if (currentUser.email === userEmail) {
+          throw new BadRequestError('Cannot alter own access level');
```

Comment:
I think this should be a ForbiddenError but I'll get around to changing it in a bit